### PR TITLE
Add unit tests for streak counter and heatmap tracker

### PR DESCRIPTION
**Pull request by shivansh31414**
This PR adds pytest-based unit tests for streak counter and heatmap tracker, covering edge cases and alert logic. Improves reproducibility and CI coverage.

### CODE REVIEW

File: daily_ninja_python/tests/unit/test_streak_counter.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import pytest
+
+# Ensure backend package imports resolve in tests
+BACKEND_ROOT = Path(__file__).resolve().parents[2] / "backend"
+if str(BACKEND_ROOT) not in sys.path:
+    sys.path.insert(0, str(BACKEND_ROOT))
+
+from run import create_app  # noqa: E402
+from app.models import db, User, Activity  # noqa: E402
+from app.api import tasks as tasks_api  # noqa: E402
```
[the remark this answers is not in the document]
Comment (shivansh31414) [reply daily_ninja_python/tests/unit/test_streak_counter.py:14]: good point out

File: daily_ninja_python/tests/unit/test_streak_counter.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+import pytest
+
+# Ensure backend package imports resolve in tests
+BACKEND_ROOT = Path(__file__).resolve().parents[2] / "backend"
+if str(BACKEND_ROOT) not in sys.path:
+    sys.path.insert(0, str(BACKEND_ROOT))
+
+from run import create_app  # noqa: E402
+from app.models import db, User, Activity  # noqa: E402
```
Comment (shivansh31414) [inline daily_ninja_python/tests/unit/test_streak_counter.py]: makes the importing less flexible
